src/components/SubNav/SubNav.tsx

```diff
@@ -0,0 +1,379 @@
+];
+export type highlightColors = typeof highlightColorsArray[number];
+
+export interface SubNavProps {
+  /** Additional class name for the `SubNav` component. */
+  className?: string;
+  /**
+   * Primary actions displayed on the left side of the SubNav.
+   * Use SubNavButton and SubNavLink components, which mirror
+   * the DS Button and Link functionalities, plus the isOutlined prop.
+   */
+  primaryActions: ({
+    highlightColor,
+    actionBackgroundColor,
+    selectedItem,
+  }: {
+    highlightColor?: string;
+    actionBackgroundColor?: string;
+    selectedItem?: string;
+  }) => React.ReactNode;
+  /**
+   * Secondary actions displayed on the right side of the SubNav.
+   * Use SubNavButton and SubNavLink components, offering additional
+   * contextual options related to the primary content.
+   */
+  secondaryActions?: ({
+    highlightColor,
+    actionBackgroundColor,
+    selectedItem,
+  }: {
+    highlightColor?: string;
+    actionBackgroundColor?: string;
+    selectedItem?: string;
+  }) => React.ReactNode;
+  /**
+   * The background color to be applied to the hover and active states
+   * of the SubNavLink and SubNavButton components.
+   * This allows for customization of the action items.
+   */
+  actionBackgroundColor?: actionBackgroundColors;
+  /**
+   * Custom color for SubNavLink, SubNavButton, and icons.
+   */
+  highlightColor?: highlightColors;
+  /**
+   * Optional unique ID for accessibility, allowing other components
+   * to reference this element.
+   */
+  id?: string;
+  /** Optional string used to identify and highlight an item. The value should
+   * match the id associated with one of the items.
+   */
+  selectedItem?: string;
+}
+
+export const isIconComponent = (childType: any): boolean => {
+  return (
+    childType.$$typeof === Symbol.for("react.forward_ref") &&
+    childType.displayName === "Icon"
+  );
+};
+
+// Function to check if any of the children is an Icon component
+export const hasIcon = (children: React.ReactNode): boolean => {
+  return React.Children.toArray(children).some((child) => {
+    if (React.isValidElement(child)) {
+      const childType = child.type as React.ComponentType<any>;
+      return isIconComponent(childType);
+    }
+    return false;
+  });
+};
+
+// Function to render content based on the screen size
+export const renderContent = (
+  children: React.ReactNode,
+  isLargerThanMobile: boolean,
+  hasIconInChildren: boolean
+): React.ReactNode => {
+  if (isLargerThanMobile) {
+    return children; // On larger screens, render everything
+  }
+
+  if (hasIconInChildren) {
+    return React.Children.map(children, (child) => {
+      if (React.isValidElement(child) && isIconComponent(child.type)) {
+        return child; // Only render the icon on smaller screens
+      }
+      return null;
+    });
+  }
+  return children; // If no icon, render text
+};
+
+export const SubNavButton: React.FC<React.PropsWithChildren<any>> = chakra(
```

Comment:
Since we want to ignore custom styles that might be passed to `SubNavButton` and `SubNavLink`, the `chakra()` function should not be used here. If it's removed, style props and the `sx` and `__css` props won't be available which is what we want.